src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -49,7 +50,9 @@ public LogicManager(Model model, Storage storage) {
     }
 
     @Override
-    public CommandResult execute(String commandText) throws CommandException, ParseException {
+    public CommandResult execute(String commandText) throws CommandException, ParseException, IOException,
```

Comment:
Great idea to use DataLoadingException, definitely adds flexibility to the exceptions thrown here